client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,118 @@
+import { BaseViewModel } from '../../site/base/base-view-model';
+import { Injectable } from '@angular/core';
+import { FileExportService } from './file-export.service';
+
+/**
+ * Handles all incoming and outgoing notify messages via {@link WebsocketService}.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class CsvExportService {
+    protected exporter = new FileExportService();
+
+    /**
+     * empty constructor TODO
+     */
+    public constructor() {}
+
+    /**
+     * Saves an array of model data to a CSV.
+     * @param data Array of Model instances to be saved
+     * @param columns Column definitions (model property; optional csv title)
+     * @param filename name of the resulting file
+     * @param options
+     */
+    public saveAsCsv(
```

Comment:
`public save<T extends BaseViewModel>(data: T[], columns: { [P in keyof T]?/* the property*/: string /* the column title*/}, filename: string, ...rest)`
Does it work to make `{ineSeparator = '\r\n', columnSeparator = ';'}` optional. If you do not want to provide values, you have to pass an empty object; this is not nice.

Reply:
```
{ lineSeparator = '\r\n', columnSeparator = ';' }:{ lineSeparator?: string; columnSeparator?: string } = {}
```
This should work (at least in es2016 this is how destructuring works with an default of 'no data submitted'; I'm testing it now)

Reply:
Perfect!